File: datasets/jl-keep-code_davinci_001_temp_0.2/HumanEval_54_same_chars.jl
"""
    Check if two words have the same characters.
    >>> same_chars('eabcdzzzz', 'dddzzzzzzzddeddabc')
    True
    >>> same_chars('abcd', 'dddddddabc')
    True
    >>> same_chars('dddddddabc', 'abcd')
    True
    >>> same_chars('eabcd', 'dddddddabc')
    False
    >>> same_chars('abcd', 'dddddddabce')
    False
    >>> same_chars('eabcdzzzz', 'dddzzzzzzzddddabc')
    False
    """
function same_chars(s0::String, s1::String)::Bool 
    if length(s0) != length(s1):
        return false
    else:
        for i in s0
            if !(i in s1)
                return false
        end
    end
    return true
end

using Test

@testset begin

candidate = same_chars;
	@test(candidate("eabcdzzzz", "dddzzzzzzzddeddabc") == true)
	@test(candidate("abcd", "dddddddabc") == true)
	@test(candidate("dddddddabc", "abcd") == true)
	@test(candidate("eabcd", "dddddddabc") == false)
	@test(candidate("abcd", "dddddddabcf") == false)
	@test(candidate("eabcdzzzz", "dddzzzzzzzddddabc") == false)
	@test(candidate("aabb", "aaccc") == false)
end

File: datasets/pl-keep-code_davinci_001_temp_0.2/HumanEval_54_same_chars.pl
# Check if two words have the same characters.
# >>> same_chars('eabcdzzzz', 'dddzzzzzzzddeddabc')
# True
# >>> same_chars('abcd', 'dddddddabc')
# True
# >>> same_chars('dddddddabc', 'abcd')
# True
# >>> same_chars('eabcd', 'dddddddabc')
# False
# >>> same_chars('abcd', 'dddddddabce')
# False
# >>> same_chars('eabcdzzzz', 'dddzzzzzzzddddabc')
# False
sub same_chars {
    my($s0, $s1) = @_;
    my %chars0 = map { $_ => 1 } split //, $s0;
    my %chars1 = map { $_ => 1 } split //, $s1;
    return scalar(keys %chars0) == scalar(keys %chars1);
}

use Test::Deep;


sub testhumaneval {
    my $candidate = \&same_chars;
        if(eq_deeply($candidate->("eabcdzzzz", "dddzzzzzzzddeddabc"),1)) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->("abcd", "dddddddabc"),1)) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->("dddddddabc", "abcd"),1)) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->("eabcd", "dddddddabc"),"")) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->("abcd", "dddddddabcf"),"")) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->("eabcdzzzz", "dddzzzzzzzddddabc"),"")) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->("aabb", "aaccc"),"")) {
        print "ok!" }else{
        exit 1;
        }
}

testhumaneval();